torch/onnx/utils.py

```diff
@@ -665,10 +669,52 @@ def _find_typename(v):
         export_modules_as_functions = module_typenames
     else:
         export_modules_as_functions = None
+
+    if export_modules_as_functions:
+        __register_attribute_hook()
+
     return export_modules_as_functions
 
 def _reset_trace_module_map():
     torch.jit._trace._trace_module_map = None
+    torch._C._jit_pass_onnx_clear_scope_records()
+
+def _get_module_attributes(module):
+    from typing import get_type_hints
+    annotations = get_type_hints(type(module))
+    base_m_annotations = get_type_hints(torch.nn.Module)
+    if annotations and base_m_annotations:
+        [annotations.pop(k, None) for k in base_m_annotations.keys()]
+    else:
+        annotations = {}
+    return {k : getattr(module, k) for k in annotations.keys()}
+
+def _register_module_forward_hook(module, hook):
+    for child in module.children():
```

Comment:
Can we use `module.modules()` to handle the recursion for us?
https://discuss.pytorch.org/t/module-children-vs-module-modules/4551